apps/meteor/server/models/raw/Subscriptions.ts

```diff
@@ -580,6 +580,12 @@ export class SubscriptionsRaw extends BaseRaw<ISubscription> implements ISubscri
 		return this.updateMany(query, update);
 	}
 
+	async setGroupE2EKeyAndOldRoomKeys(_id: string, key: string, oldRoomKeys: ISubscription['oldRoomKeys']): Promise<UpdateResult> {
+		const query = { _id };
+		const update = { $set: { E2EKey: key, ...(oldRoomKeys && { oldRoomKeys }) } };
```

Comment:
The rotation doesn't happen at the DB, it happens on the `resetRoomKey` func. This was to avoid having to find the subscription again 😬 , so we pull the data, rotate the keys and set the full array again.